### PR TITLE
Ensure working JVM before enabling Jib actions to avoid hangs

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes: #5493
Fixes: #5380

**Description**
macOS provides facades for the common Java executables (e.g., `java`, `javac`) that attempt to find and invoke the corresponding commands from a suitable Java VM (normally found in `/Library/Java/JavaVirtualMachines`).  When no Java VM is available, these facades error:
```console
$ java -version
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
```

But using a Maven Wrapper script (`mvnw`) instead results in a hang!  Tracing through the script shows:
```
$ sh -x mvnw
...
WRAPPER_LAUNCHER=org.apache.maven.wrapper.MavenWrapperMain
+ exec /usr/bin/java -Xmx1g -classpath /Users/bdealwis/Projects/Skaffold/repo-skaffold/integration/examples/jib/.mvn/wrapper/maven-wrapper.jar -Dmaven.home=/Users/bdealwis/Projects/Skaffold/repo-skaffold/integration/examples -Dmaven.multiModuleProjectDirectory=/Users/bdealwis/Projects/Skaffold/repo-skaffold/integration/examples/jib org.apache.maven.wrapper.MavenWrapperMain
```
---
This PR adds a check that a working JVM exists before using Jib.

### CODE REVIEW

File: pkg/skaffold/build/jib/jvm.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+var (
+	// JVMFound is true if a Java VM was found and works.
+	JVMFound bool
+)
+
+func init() {
```
Comment (MarlonGamez) [inline pkg/skaffold/build/jib/jvm.go:32]: What are the benefits of doing this in an init function vs doing this with a `sync.Once`? Would this run for basically all skaffold invocations?

**Review by MarlonGamez — COMMENTED** [review]
Seems good! Just have one question

File: pkg/skaffold/build/jib/jvm.go (hunk 1 above)
Comment (briandealwis) [reply pkg/skaffold/build/jib/jvm.go:32]: It simplified the implementation slightly.  A package `init()` should only run whenever the package is loaded, but as I've discovered when putting a `panic()` into this function, we touch the Jib package early on:
```console
$ ../../../out/skaffold build -v debug
panic: resolveJvm() called

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/jib.resolveJVM(0x1400039a420)
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/pkg/skaffold/build/jib/jvm.go:40 +0x38
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/jib.init.0()
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/pkg/skaffold/build/jib/jvm.go:33 +0x20
```

I'll rewrite it to use a `sync.Once`.

File: pkg/skaffold/build/jib/jvm.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+// JVMFound returns true if a Java VM was found and works.
+func JVMFound() bool {
+	resolveJVMOnce.Do(func() {
+		jvmPresent = resolveJVM()
+	})
+	return jvmPresent
+}
+
+// resolveJVMForInit returns true if a Java VM was found and works.  It is intended for
```
Comment (MarlonGamez) [inline pkg/skaffold/build/jib/jvm.go:42]:
```suggestion
// resolveJVM returns true if a Java VM was found and works.  It is intended for
```

**Review by MarlonGamez — COMMENTED** [review]
Looking good, just noticed a tiny inconsistency

**Comment by briandealwis** [comment]
I hit hanging-tests again this morning when testing a different change on my M1!

PTAL @MarlonGamez: I tweaked the implementation again and this is now working on all platforms.

**Review by nkubala — APPROVED** [review]

**Pull request merged**